syft/tensor.py

```diff
@@ -1372,6 +1374,30 @@ def mm(self, tensor2):
 
         return syft.mm(self, tensor2)
 
+    def stride(self, dim=None):
+        """Returns the jump necessary to go from one element to the next one in the specified dimension dim."""
+        if self.encrypted:
+            return NotImplemented
+        tensor = np.array(self.data)
+        shp = tensor.shape
+        ndim = len(shp)
+
+        def calculation(ndim, shp):
+            out = []
+            for i in range(1, ndim):
+                out.append(reduce(mul, shp[i:]))
+            out.append(1)
+            out = tuple(out)
+            return out
+        if dim is None:
+            out = calculation(ndim, shp)
+            return out
+        elif ((dim >= 0) and (dim < ndim)) or ((dim < 0) and (dim >= -ndim)):
+            out = calculation(ndim, shp)
+            return out[dim]
+        else:
+            raise RuntimeError("dimension out of range of {}D tensor".format(ndim))
```

Comment:
Runtime error!! I guess not.
We can do a simple `Error` or better `ValueError` which suits this situation.
Look into these. `RuntimeError` is definitely not the one we should be using.

Reply:
Pytorch used Runtime error for this issue.